### PR TITLE
add benchmarks for permissions

### PR DESCRIPTION
**Pull request by vedhavyas**

### CODE REVIEW

File: pallets/permissions/src/lib.rs [hunk 1]
```diff
@@ -233,10 +247,11 @@ impl<T: Config> Pallet<T> {
 					roles.rm(role).map_err(|_| Error::<T>::WrongParameters)?;
 
 					if roles.empty() {
-						Ok(Permission::<T>::remove(who, location))
+						Permission::<T>::remove(who, location);
 					} else {
-						Ok(Permission::<T>::insert(who, location, roles))
+						Permission::<T>::insert(who, location, roles);
 					}
+					Ok(())
```
Comment (mustermeiszer) [inline pallets/permissions/src/lib.rs:254]: Why this change?

File: pallets/permissions/src/mock.rs [hunk 2]
```diff
@@ -116,11 +116,12 @@ impl Properties for Storage {
 					self.org.remove(OrgStorage::HEAD_OF_SAUBERMACHING)
 				}
 			},
-		})
+		};
+		Ok(())
```
Comment (mustermeiszer) [inline pallets/permissions/src/mock.rs:120]: Why this change?

File: pallets/permissions/src/mock.rs [hunk 3]
```diff
@@ -116,11 +116,12 @@ impl Properties for Storage {
...
 	}
 
 	fn add(&mut self, property: Self::Property) -> Result<(), ()> {
-		Ok(match property {
```
Comment (mustermeiszer) [inline pallets/permissions/src/mock.rs:123]: See above.

File: pallets/permissions/src/lib.rs [hunk 4]
```diff
@@ -167,28 +185,26 @@ impl<T: Config> Pallet<T> {
...
+			});
 
 			if T::Editors::contains(&(who, has_role, location, role)) {
-				return Ok(());
+				return Ok(Who::Editor);
 			}
 		}
 
-		Self::ensure_admin(origin)
+		Self::ensure_admin(origin).map(|_| Who::Root)
```
Comment (mustermeiszer) [inline pallets/permissions/src/lib.rs:203]: I think with the approach of the weights, we should early return here in any case. As a `Origin::signed(who)`, where `who` is not contained in the `Editors` will get the weight for the `Who::Root`, which is less and not correct. 

Or we move the admin check to the top.

**Review by mustermeiszer — APPROVED** [review]
LGTM! Just one thing were I think the weights are calculated incorrectly. And some minor remarks.

File: pallets/permissions/src/mock.rs (hunk 2 above)
Comment (vedhavyas) [reply pallets/permissions/src/mock.rs:120]: https://rust-lang.github.io/rust-clippy/master/index.html#unit_arg

File: pallets/permissions/src/mock.rs (hunk 3 above)
Comment (vedhavyas) [reply pallets/permissions/src/mock.rs:123]: https://rust-lang.github.io/rust-clippy/master/index.html#unit_arg

File: pallets/permissions/src/lib.rs (hunk 1 above)
Comment (vedhavyas) [reply pallets/permissions/src/lib.rs:254]: rust-lang.github.io/rust-clippy/master/index.html#unit_arg

File: pallets/permissions/src/lib.rs (hunk 4 above)
Comment (mustermeiszer) [reply pallets/permissions/src/lib.rs]: Nope, you are right. Then nevermind.

Comment (mustermeiszer) [reply pallets/permissions/src/lib.rs]: And what if you are Admin but not editor? Can a signed extrinsic pass the admin check? Yes, right? Then this edge case would get the wrong weight.

Comment (vedhavyas) [reply pallets/permissions/src/lib.rs]: I have refactored it a lil bit. take another look ?

**Review by mustermeiszer — APPROVED** [review]
LGTM!
